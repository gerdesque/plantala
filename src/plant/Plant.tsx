import React from 'react';
import { Typography } from '@material-ui/core';

export interface IPlant {
  active: boolean
  name: string
  source: string
  amount?: number
  description?: string
  order ?: number
  rotation?: number
  scale?: number
  size?: number
}

export default function Plant({name, source, description}: IPlant) {

  return (
    <div className="plant">
      <Typography variant="h2">
        {name}
      </Typography>
    </div>
  )
}
