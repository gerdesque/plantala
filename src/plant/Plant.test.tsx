import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Plant from './Plant';
import Typography from '@material-ui/core/Typography';

describe('Header', () => {

  let shallow: any;
  let wrapper: any;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Plant name='' source='' />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div /> container', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the title "Plantala"', () => {
    expect(wrapper.find(Typography).text()).toEqual('');
  });
  

});
