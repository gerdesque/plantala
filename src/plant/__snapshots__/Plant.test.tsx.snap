// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="plant"
>
  <WithStyles(ForwardRef(Typography))
    variant="h2"
  />
</div>
`;
